stub_uploader/metadata.py

```diff
@@ -350,9 +350,11 @@ def verify_external_req_stubs_require_its_runtime(
 
     runtime_req_name = EXTERNAL_RUNTIME_REQ_MAP.get(req.name, req.name)
 
-    if not runtime_in_upstream_requires(
-        req, data
-    ) and not runtime_in_upstream_sdist_requires(req, data):
+    if (
+        runtime_req_name != upstream_distribution  # Allow `types-foo` to require `foo`
```

Comment:
if you want me to hide this check behind a function for consistency, I'd call it `runtime_is_upstream`.